tests/testthat/test-detect.r

```diff
@@ -36,6 +36,10 @@ test_that("str_starts works", {
   # Special typing of patterns.
   expect_true(str_starts("ab", fixed("A", ignore_case = TRUE)))
   expect_true(str_starts("ab", regex("A", ignore_case = TRUE)))
+
+  # Or operators are respected
+  expect_true(str_starts("ab", regex("B|A", ignore_case = TRUE)))
```

Comment:
Also I don’t think you should using `ignore_case` as that’s unrelated to this isse